### PR TITLE
fix(compute): fixed compute_reservation_create_shared sample and test to use mocked client

### PR DESCRIPTION
**Pull request by TetyanaYahodska**
## Description 

In this PR I made improvements to compute_reservation_create_shared and its test. We now use a similar approach to the mock client as we do for the other samples in the compute package.

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist

- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Comment by minherz** [comment]
We do not review draft PRs. If this PR is ready for review and merge, please convert it to active PR.

**Comment by iennae** [comment]
@gemini review

File: compute/cloud-client/src/main/java/compute/reservation/CreateSharedReservation.java [hunk 1]
```diff
@@ -57,53 +52,57 @@ public static void main(String[] args)
...
-  public void createSharedReservation(
-      String projectId, String zone,
-      String reservationName, String instanceTemplateUri, int vmCount)
-      throws ExecutionException, InterruptedException, TimeoutException {
+  public static Operation.Status createSharedReservation(
+          String projectId, String zone,
+          String reservationName, String instanceTemplateUri, int vmCount)
+          throws ExecutionException, InterruptedException, TimeoutException, IOException {
 
     ShareSettings shareSettings = ShareSettings.newBuilder()
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/reservation/CreateSharedReservation.java:65]: Move the ShareSettings within the try block as well.

File: compute/cloud-client/src/main/java/compute/reservation/CreateSharedReservation.java [hunk 2]
```diff
@@ -57,53 +52,57 @@ public static void main(String[] args)
...
-    creator.createSharedReservation(projectId, zone, reservationName, instanceTemplateUri, vmCount);
+    createSharedReservation(projectId, zone, reservationName, instanceTemplateUri, vmCount);
   }
 
   // Creates a shared reservation with the given name in the given zone.
-  public void createSharedReservation(
-      String projectId, String zone,
-      String reservationName, String instanceTemplateUri, int vmCount)
-      throws ExecutionException, InterruptedException, TimeoutException {
+  public static Operation.Status createSharedReservation(
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/reservation/CreateSharedReservation.java:60]: Can we modify the return type to just Status and import the package?

`public static Status createSharedReservation()`

File: compute/cloud-client/src/main/java/compute/reservation/CreateSharedReservation.java [hunk 3]
```diff
@@ -57,53 +52,57 @@ public static void main(String[] args)
         "projects/%s/global/instanceTemplates/YOUR_INSTANCE_TEMPLATE_NAME", projectId);
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/reservation/CreateSharedReservation.java:52]:
```suggestion
        "projects/%s/global/instanceTemplates/%s", projectId, "YOUR_INSTANCE_TEMPLATE_NAME");
```

**Review by Sita04 — CHANGES_REQUESTED** [review]

File: compute/cloud-client/src/main/java/compute/reservation/CreateSharedReservation.java (hunk 2 above)
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/reservation/CreateSharedReservation.java:60]: Thank you! Fixed

File: compute/cloud-client/src/main/java/compute/reservation/CreateSharedReservation.java (hunk 1 above)
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/reservation/CreateSharedReservation.java:65]: Thank you! Fixed

**Review by Sita04 — APPROVED** [review]

**Pull request merged**